src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -502,12 +506,11 @@ public final Response getEventBookingByEventId(@Context final HttpServletRequest
         try {
             RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(request);
 
-            List<EventBookingDTO> eventBookings = bookingManager.getBookingsByEventId(eventId);
+            List<EventBookingDTO> eventBookings = this.mapper.mapAsList(bookingManager.getBookingsByEventId(eventId), EventBookingDTO.class);
 
             // Event leaders are only allowed to see the bookings of connected users
-            if (Role.EVENT_LEADER.equals(currentUser.getRole())) {
-                eventBookings = userAssociationManager.filterUnassociatedRecords(
-                        currentUser, eventBookings, booking -> booking.getUserBooked().getId());
+            if (Arrays.asList(Role.STUDENT, Role.TEACHER, Role.EVENT_LEADER, Role.CONTENT_EDITOR).contains(currentUser.getRole())) {
```

Comment:
Pretty sure this shouldn't contain Role.STUDENT?

Reply:
Likely not. This wouldn't prevent you from seeing your own bookings now, would it?